packages/pds/src/app-view/api/app/bsky/feed/getTimeline.ts

```diff
@@ -111,3 +118,56 @@ export default function (server: Server, ctx: AppContext) {
     },
   })
 }
+
+const ensureReadAfterWrite = async (
+  ctx: AppContext,
+  requester: string,
+  res: ApiRes<OutputSchema>,
+): Promise<OutputSchema> => {
+  const clock = getClock(res.headers)
+  if (!clock) return res.data
+  const localSrvc = ctx.services.local(ctx.db)
+  const local = await localSrvc.getRecordsSinceClock(requester, clock, [
+    ids.AppBskyFeedPost,
+  ])
+  const formatted = await findAndFormatPostsToInsert(
+    localSrvc,
+    local.posts,
+    res.data.feed,
+  )
+  if (formatted.length === 0) return res.data
+  const feed = [...res.data.feed]
+  for (const post of formatted) {
+    if (post === null) continue
```

Comment:
If I'm understanding correctly, I don't think `null`s should be able to show-up here 👍 Maybe the filtering of nulls was added late to `findAndFormatPostsToInsert()`.

Reply:
ah yup that's right 👍